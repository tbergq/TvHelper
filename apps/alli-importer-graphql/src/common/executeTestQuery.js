// @flow

import { graphql } from 'graphql';

import schema from '../Schema';

const createContext = () => ({});

export default function executeTestQuery(
  query: string,
  variables: ?{ ... },
  request: ?{ ... } = {},
): Promise<{ ... }> {
  // $FlowExpectedError: Ok for testing purposes
  return graphql(schema, query, null, createContext(request), variables);
}
