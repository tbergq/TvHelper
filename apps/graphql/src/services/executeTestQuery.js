// @flow

import { graphql } from 'graphql';

import schema from '../Schema';
import createContext from './createGraphqlContext';

export default function executeTestQuery(
  query: string,
  variables: ?{ ... },
  request: ?{ ... } = {},
): Promise<{ ... }> {
  // $FlowExpectedError: Ok for testing purposes
  return graphql(schema, query, null, createContext(request), variables);
}
