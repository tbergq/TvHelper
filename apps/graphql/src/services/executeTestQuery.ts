import { graphql } from 'graphql';

import schema from '../Schema';
import createContext from './createGraphqlContext';

export default function executeTestQuery(
  query: string,
  variables: {} | null | undefined,
  request: {} | null | undefined = {},
): Promise<{}> {
  // @ts-ignore: Ok for testing purposes
  return graphql(schema, query, null, createContext(request), variables);
}
