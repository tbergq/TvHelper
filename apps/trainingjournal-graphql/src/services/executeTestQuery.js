// @flow

import { graphql } from 'graphql';

import schema from '../Schema';

export default function executeTestQuery(query: string, variables: ?{ ... }): Promise<{ ... }> {
  return graphql(schema, query, null, null, variables);
}
