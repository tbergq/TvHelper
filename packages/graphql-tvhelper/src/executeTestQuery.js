// @flow

import { graphql as originalGraphQL, GraphQLSchema, GraphQLObjectType } from 'graphql';

import SearchTvShow from './search/queries/SearchTvShow';

const RootQuery = new GraphQLObjectType({
  name: 'RootQuery',
  description: 'Root Query',
  fields: {
    searchTvShow: SearchTvShow,
  },
});

const schema = new GraphQLSchema({
  query: RootQuery,
});

/**
 * Executes GraphQL query using our schema. This function should
 * be used only during testing because it uses faked context.
 */
export default function executeTestQuery(
  query: string,
  variables: ?{ ... },
  context: ?{ ... },
): Promise<Object> {
  return originalGraphQL(schema, query, null, context, variables);
}
