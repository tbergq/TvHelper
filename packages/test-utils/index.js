// @flow

export { default as withMongoDb } from './src/useMongoDb';
