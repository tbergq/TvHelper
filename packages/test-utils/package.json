{
  "name": "@tbergq/test-utils",
  "version": "0.0.1",
  "private": true,
  "main": "index.js",
  "dependencies": {
    "mongodb-memory-server": "^5.2.0",
    "mongoose": "^5.6.9"
  }
}
