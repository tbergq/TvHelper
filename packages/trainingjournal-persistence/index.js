// @flow strict

import Exercise from './src/dataObjects/Exercise';

// Repositories
export { default as UserRepository } from './src/repositories/UserRepository';
export { default as ExerciseRepository } from './src/repositories/ExerciseRepository';
export { default as ProgramRepository } from './src/repositories/ProgramRepository';

// Connection
export { default as trainingJournalConnection } from './src/connection';

// dataObject

export type ExerciseType = typeof Exercise;
