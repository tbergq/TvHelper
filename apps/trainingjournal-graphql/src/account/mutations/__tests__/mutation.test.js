// @flow

import { connection, UserRepository } from '@tbergq/trainingjournal-persistence';
import { generateTestsFromFixtures } from '@adeira/test-utils';
import path from 'path';
import { nullthrows } from '@adeira/js';
import Mockdate from 'mockdate';

import executeTestQuery, { getOperationName } from '../../../services/executeTestQuery';

Mockdate.set(new Date(2020, 0, 1, 0, 0, 0, 0));

jest.mock('@adeira/graphql-relay', () => {
  const original = jest.requireActual('@adeira/graphql-relay');
  return {
    toGlobalId: () => original.toGlobalId('User', 1),
  };
});

const user = {
  username: 'tito',
  email: 'tito@bonito.com',
  password: 'password',
};

const inputs = {
  createUser: { input: user },
  login: { username: 'tita', password: 'password' },
};

beforeEach(async () => {
  await UserRepository.createUser({
    username: 'tita',
    password: 'password',
    email: 'tita@bebita.com',
  });
});

afterEach(async () => {
  await connection.query('delete from users;');
});

generateTestsFromFixtures(path.join(__dirname, '__fixtures__'), (input) => {
  const key = nullthrows(getOperationName(input), 'Should exist');
  return executeTestQuery(input, inputs[key]);
});
