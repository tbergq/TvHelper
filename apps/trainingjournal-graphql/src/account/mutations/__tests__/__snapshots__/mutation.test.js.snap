// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches expected output: createUser.graphql 1`] = `
~~~~~~~~~~ INPUT ~~~~~~~~~~
mutation createUser($input: CreateUserInput!) {
  createUser(input: $input) {
    ... on User {
      username
      email
    }
    ... on CreateUserError {
      message
      reason
    }
  }
}
~~~~~~~~~~ OUTPUT ~~~~~~~~~~
{
  "data": {
    "createUser": {
      "username": "tito",
      "email": "tito@bonito.com"
    }
  }
}
`;

exports[`matches expected output: login.graphql 1`] = `
~~~~~~~~~~ INPUT ~~~~~~~~~~
mutation login($username: String!, $password: String!) {
  login(username: $username, password: $password) {
    token
    success
  }
}

~~~~~~~~~~ OUTPUT ~~~~~~~~~~
{
  "data": {
    "login": {
      "token": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6IlZYTmxjam94IiwidXNlcm5hbWUiOiJ0aXRhIiwiZW1haWwiOiJ0aXRhQGJlYml0YS5jb20iLCJpYXQiOjE1Nzc4MzY4MDAsImV4cCI6MTYwOTM5NDQwMCwiaXNzIjoidGJlcmdxLXRyYWluaW5nam91cm5hbC5oZXJva3VhcHAuY29tIn0.KSOGublkXmCLANbv5DmgygbWqTBC6mAbQz1nMiCdvyA",
      "success": true
    }
  }
}
`;
